perceval/components/unitary_components.py

```diff
@@ -187,20 +188,25 @@ class PS(ACircuit):
     """Phase shifter"""
     DEFAULT_NAME = "PS"
 
-    def __init__(self, phi):
+    def __init__(self, phi, max_error = 0):
         super().__init__(1)
-        self._phi = self._set_parameter("phi", phi, 0, 2*sp.pi)
+        self._phi = self._set_parameter("phi", phi, 0, 2*math.pi)
+        self._max_error = self._set_parameter("max_error", max_error, 0, math.pi)
 
     def _compute_unitary(self, assign=None, use_symbolic=False):
         self.assign(assign)
         if use_symbolic:
-            return Matrix([[sp.exp(self._phi.spv*sp.I)]], True)
+            err = self._max_error.spv*random.uniform(-1, 1)
+            return Matrix([[sp.exp((self._phi.spv + err)*sp.I)]], True)
         else:
-            return Matrix([[math.cos(float(self._phi)) + 1j * math.sin(float(self._phi))]], False)
+            err = float(self._max_error)*random.uniform(-1, 1)
+            return Matrix([[math.cos(float(self._phi) + err) + 1j * math.sin(float(self._phi) + err)]], False)
```

Comment:
Just for readability, create a phase variable that would be `float(self._phi) + err`